src/System.Memory/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Date.cs

```diff
@@ -0,0 +1,135 @@
+                        {
+                            value = default;
+                            bytesConsumed = 0;
+                            return false;
+                        }
+
+                        switch (kind)
+                        {
+                            case DateTimeKind.Local:
+                                value = dateTimeOffset.LocalDateTime;
+                                break;
+                            case DateTimeKind.Utc:
+                                value = dateTimeOffset.UtcDateTime;
+                                break;
+                            case DateTimeKind.Unspecified:
+                                value = dateTimeOffset.DateTime;
+                                break;
+                            default:
+                                Debug.Assert(false, "Unexpected DateTimeKind: " + kind);
+                                throw new InvalidOperationException();
+                        }
+                        return true;
+                    }
+
+                case default(char):
+                case 'G':
+                    return TryParseDateTimeG(text, out value, out _, out bytesConsumed);
+
+                default:
+                    throw new FormatException(SR.Argument_BadFormatSpecifier);
+            }
+        }
+
+        /// <summary>
+        /// Parses a SByte at the start of a Utf8 string.
+        /// </summary>
+        /// <param name="text">The Utf8 string to parse</param>
+        /// <param name="value">Receives the parsed value</param>
+        /// <param name="bytesConsumed">On a successful parse, receives the length in bytes of the substring that was parsed </param>
+        /// <param name="standardFormat">Expected format of the Utf8 string</param>
+        /// <returns>
+        /// true for success. "bytesConsumed" contains the length in bytes of the substring that was parsed.
+        /// false if the string was not syntactically valid or an overflow or underflow occurred. "bytesConsumed" is set to 0. 
+        /// </returns>
+        /// <remarks>
+        /// Formats supported:
+        ///     G  (default)  05/25/2017 10:30:15
+        ///     R             Tue, 03 Jan 2017 08:08:05 GMT       (RFC 1123)
+        ///     l             tue, 03 jan 2017 08:08:05 gmt       (Lowercase RFC 1123)
+        ///     O             2017-06-12T05:30:45.7680000-07:00   (Round-trippable)
+        /// </remarks>
+        /// <exceptions>
+        /// <cref>System.FormatException</cref> if the format is not valid for this data type.
+        /// </exceptions>
+        public static bool TryParse(ReadOnlySpan<byte> text, out DateTimeOffset value, out int bytesConsumed, char standardFormat = default)
+        {
+            switch (standardFormat)
+            {
+                case 'R':
+                case 'l':
+                    return TryParseDateTimeOffsetR(text, out value, out bytesConsumed);
```

Comment:
Are we ok with passing in 'R' and then successfully parsing lower cased date time?

Reply:
We've generally been "case-insensitive" for parsing but given that this is a RFC wire format, maybe an exception is warranted. Waiting to see what others have to say on this.

Reply:
I've made them 'strict case' for now.